tests/test.ZoneeventWatcherOverflow.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2016, Joyent, Inc.
+ */
+
+var execFile = require('child_process').execFile;
+
+var test = require('tape');
+var vasync = require('vasync');
+var vmadm = require('vmadm');
+
+var common = require('./common');
+var mocks = require('./mocks');
+var ZoneeventWatcher = require('../lib/watchers/zoneevent-watcher');
+
+var createdVms = [];
+var debugData = {totalEvents: 0, totalLen: 0};
+var smartosImageUUID;
+var watcher;
+var vmEvents = {};
+
+
+// "CONFIG"
+var CHARS_PER_LINE = 200;  // at least this many chars in a zoneevent line
+var STRAGGLER_WAIT = 5000; // wait this many ms for events after delete
+var OVERHEAD_EVENTS = 15;  // events we expect for create/delete
+var RESTART_EVENTS = 9;    // events we expect for each restart
+var NUM_RESTARTS_PER_VM = 25;
+var NUM_VMS = 10;
+
+
+/*
+ * This test exists to confirm that AGENT-987 is fixed.
+ *
+ * Basically it proves that we can successfully handle at least (NUM_VMS *
+ * NUM_RESTARTS_PER_VM * 9) + (NUM_VMS * 15) events.
+ *
+ * To test this, we create NUM_VMS VMs and reboot them in parallel
+ * NUM_RESTARTS_PER_VM times.
+ *
+ * This should result in that number of start + stop events for each of them
+ * along with 7 other intermediate events per restart and 15 events per VM for
+ * the create/destroy.
+ *
+ */
+
+test('find SmartOS image', function _test(t) {
+    common.testFindSmartosImage(t, function _findSmartosCb(err, latest) {
+        t.ifError(err, 'find SmartOS Image');
+        if (err) {
+            throw new Error('Cannot continue without SmartOS Image');
+        }
+        smartosImageUUID = latest;
+        t.end();
+    });
+});
+
+test('starting ZoneeventWatcher', function _test(t) {
+    function _onVmUpdate(vmUuid, updateType, updateObj) {
+        if (!vmEvents.hasOwnProperty(vmUuid)) {
+            vmEvents[vmUuid] = {start: 0, stop: 0, other: 0};
+        }
+
+        if (updateObj.zone_state === 'running') {
+            vmEvents[vmUuid].start++;
+        } else if (updateObj.zone_state === 'stopped') {
+            vmEvents[vmUuid].stop++;
+        } else {
+            vmEvents[vmUuid].other++;
+        }
+    }
+
+    watcher = new ZoneeventWatcher({
+        debugEvents: true,
+        highWaterMark: 32, // default is 16k! takes too long to get there.
```

Comment:
Should the `highWaterMark` constant be defined the same way the other constants (such as `OVERHEAD_EVENTS `) are defined and assert that the number of events expected is > than `highWaterMark`?